### ISSUE TITLE
v8.1.0.0 has problem with config Include directive

### ISSUE THREAD

**Issue by tolgaulas** [PowerShell/Win32-OpenSSH#1762]
**Environement** 
Windows 10 and
PS C:\WINDOWS\system32> Get-Command ssh*exe
CommandType     Name                                               Version    Source
-----------     ----                                               -------    ------
Application     ssh.exe                                            8.1.0.0    C:\Program Files\OpenSSH-Win64\ssh.exe
Application     ssh-add.exe                                        8.1.0.0    C:\Program Files\OpenSSH-Win64\ssh-add...
Application     ssh-agent.exe                                      8.1.0.0    C:\Program Files\OpenSSH-Win64\ssh-age...
Application     sshd.exe                                           8.1.0.0    C:\Program Files\OpenSSH-Win64\sshd.exe
Application     ssh-keygen.exe                                     8.1.0.0    C:\Program Files\OpenSSH-Win64\ssh-key...
Application     ssh-keyscan.exe                                    8.1.0.0    C:\Program Files\OpenSSH-Win64\ssh-key...
Application     ssh-shellhost.exe                                  8.1.0.0    C:\Program Files\OpenSSH-Win64\ssh-she...

**Configs**
C:\\Users\\auser/.ssh/config
-----------------------------
Include theinclude

C:\\Users\\auser/.ssh/theinclude
-----------------------------
Host sample_host
 HostName 127.0.0.1
 User auser
 IdentityFile ~/.ssh/somekey.pub

**Result**
C:\>ssh sample_host -vvv
OpenSSH_for_Windows_8.1p1, LibreSSL 2.9.2
debug1: Reading configuration data C:\\Users\\auser/.ssh/config
debug3: C:\\Users\\auser/.ssh/config line 1: Including file C:/Users/auser/.ssh/theinclude depth 0 (parse only)
debug1: Reading configuration data C:/Users/auser/.ssh/theinclude
debug3: Failed to open file:C:/ProgramData/ssh/ssh_config error:2
debug2: resolving "sample_host" port 22
ssh: Could not resolve hostname sample_host: No such host is known.

**Comment by vthiebaut10** [comment]
I couldn't reproduce this error. 
I got the following logs when I tried to replicate the problem:

```
OpenSSH_for_Windows_8.1p1, LibreSSL 2.9.2
debug1: Reading configuration data C:\\Users\\auser/.ssh/config
debug3: C:\\Users\\auser/.ssh/config line 1: Including file C:\\Users\\auser/.ssh/theinclude depth 0
debug1: Reading configuration data C:\\Users\\auser/.ssh/theinclude
debug1: C:\\Users\\auser/.ssh/theinclude line 1: Applying options for sample_host
debug3: Failed to open file:C:/ProgramData/ssh/ssh_config error:2
debug2: resolve_canonicalize: hostname 127.0.0.1 is address
debug2: ssh_connect_direct
debug1: Connecting to sample_host [127.0.0.1] port 22.
debug1: Connection established.
...
```
The third line of the log differs from the one in the description of the issue in that it doesn't have a "(parse only)" at the end of the line. The remaining of the log also indicates that the options were correctly being applied for sample_host and the connection was established successfully. 
I couldn't find a way to produce the same result of the issue even after going through the source code and identifying the parts of the code that could potentially cause this problem. I think I need more details on the configuration setup of the machine when the error was produced to be able to debug it.

**Comment by mozzieongit** [comment]
Hi,
I seem to have had the same issue and found this thread. So for other people finding this, I came across a post on superuser.com solving this: https://superuser.com/questions/1162387/whats-wrong-with-my-openssh-include-directive

TL;DR
The Include Statement cannot follow a Host or (restrictive) Match directive as it is then used **within** that directive. To solve this, either put the Include directive to the top of the file or open a new Match block like so:
```
Match all
Include config.d/*
# or other locations
```

Have a nice day everyone.
